.github/workflows/slsa-goreleaser.yml

```diff
@@ -0,0 +1,35 @@
+name: SLSA go releaser
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - "*"
+
+permissions: read-all
+
+jobs:
+  # Generate ldflags dynamically.
+  args:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.ldflags.outputs.version }}
+    steps:
+      - id: checkout
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+        with:
+          fetch-depth: 0
+      - id: ldflags
+        run: |
+          echo "::set-output name=version::$(git describe --tags --always --dirty | cut -c2-)"
+
+  # Trusted builder.
+  build:
+    permissions:
+      id-token: write
+      contents: write
```

Comment:
This needs `contents: write` in order to attach to Releases right? It's not going to push anything back to the repo I hope?